src/braket/experimental/autoqasm/program/program.py

```diff
@@ -356,26 +357,44 @@ def _free_symbol_names(expr: FreeParameterExpression) -> Iterable[str]:
         return sorted([str(s) for s in expr._expression.free_symbols if isinstance(s, Symbol)])
 
     def register_parameter(
-        self, parameter_name: str, parameter_type: Union[float, int, bool] = float
+        self,
+        parameter_name: str,
+        parameter_type: Union[float, int, bool] = float,
     ) -> None:
         """Register an input parameter if it has not already been registered.
 
         Args:
             parameter_name (str): The name of the parameter to register with the program.
             parameter_type (Union[float, int, bool]): The type of the parameter to register
                 with the program. Default: float.
+
+        Raises:
+            NotImplementedError: If the parameter type is not supported.
         """
         # TODO (#814): add type validation against existing inputs
         if parameter_name not in self._free_parameters:
-            if parameter_type == float:
-                var_class = oqpy.FloatVar
-            elif parameter_type == int:
-                var_class = oqpy.IntVar
-            elif parameter_type == bool:
-                var_class = oqpy.BoolVar
-            else:
+            aq_type = aq_types.map_parameter_type(parameter_type)
+            if aq_type not in [oqpy.FloatVar, oqpy.IntVar, oqpy.BoolVar]:
                 raise NotImplementedError(parameter_type)
-            self._free_parameters[parameter_name] = var_class("input", name=parameter_name)
+            self._free_parameters[parameter_name] = aq_type("input", name=parameter_name)
+
+    def register_output(
```

Comment:
For consistency, is it worth calling this `register_output_parameter`, and the one above `register_input_parameter`?